### PR TITLE
py-callmonitor: added new package

### PR DESCRIPTION
**Pull request by DaxLynch**
> A Simple Tool to Monitor and Log Function Calls

@eugeneswalker @elliottslaughter

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-callmonitor/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    """A Simple Tool to Monitor and Log Function Calls"""
+
+    homepage = "https://github.com/JBlaschke/call-monitor"
+    pypi = "callmonitor/callmonitor-0.3.7.tar.gz"
+
+    maintainers("DaxLynch")
+
+    version("0.3.7", sha256="11bacfe5940c3f6aff223e8e761b033d540542b4d738f7fef38cd923b3be0cbc")
+
+    depends_on("py-setuptools", type="build")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-callmonitor/package.py:19]: `py-numpy` is missing as dependency (see `install_requires` in `setup.py`

Comment (eugeneswalker) [reply var/spack/repos/builtin/packages/py-callmonitor/package.py:19]: Done!

File: var/spack/repos/builtin/packages/py-callmonitor/package.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    depends_on("py-numpy")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-callmonitor/package.py]:
```suggestion
    depends_on("py-numpy", type=("build", "run"))
```

**Review by manuelakuhn — APPROVED** [review]

**Review by eugeneswalker — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-callmonitor/package.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-callmonitor/package.py:1]:
```suggestion
# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the homepage and version sha256.  Please update the ending copyright year.

**Comment by tldahlgren** [comment]
Added @adamjstewart as reviewer in case he wants to review as well.

**Review by adamjstewart — APPROVED** [review]
Copyright year isn't important since tests pass.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**